lib/bank_account/account.rb

```diff
@@ -0,0 +1,29 @@
+module BankAccount
+  class Account
+    attr_reader :balance
+
+    def initialize(id, initial_balance)
+      # cannot start a bank account with a negative balance
+      raise ArgumentError, "Invalid amount." if initial_balance < 0
+      @balance = initial_balance
+      @account_id = id
+    end
+
+    def withdraw(amount)
```

Comment:
You could DRY up this method a bit by leveraging a guard clause like...

```ruby
def withdraw(amount)
  # the puts is nice, but not requried. In the future, we'd use the `warn` method
  return @balance if amount < @balance

  # this expression returns the result of the operation,
  # and since it's the last line of the method, that result is the implicit return value
  @balance -= amount
end
```